com.unity.render-pipelines.universal/Runtime/UniversalProfiling.cs

```diff
@@ -0,0 +1,253 @@
+        // Which show the GPU and the CPU time for executing the command buffer.
+        // Also creates "Inl_" CPU time scope outside of the command buffer.
+        public static ProfilingScope GetGpuCpuScope(CommandBuffer cmd, ProfilingSampler scopeSampler)
+        {
+            return new ProfilingScope(cmd, scopeSampler);
+        }
+
+        // Creates an enum named "immediate" CPU scope.
+        // Use 'URPProfileId' enum type to track the scope in the performance test framework.
+        public static ProfilingScope GetCpuScope<TEnum>(TEnum scopeId)
+            where TEnum : Enum
+        {
+            return new ProfilingScope(null, ProfilingSampler.Get(scopeId));
+        }
+
+        // Creates an enum named GPU + CPU scope into a command buffer.
+        // Use 'URPProfileId' enum type to track the scope in the performance test framework.
+        public static ProfilingScope GetGpuCpuScope<TEnum>(CommandBuffer cmd, TEnum scopeId)
+            where TEnum : Enum
+        {
+            return new ProfilingScope(cmd, ProfilingSampler.Get(scopeId));
+        }
+
+        // Creates a cached string named "immediate" CPU scope
+        public static ProfilingScope GetCpuScope(string scopeName)
+        {
+            return new ProfilingScope(null, TryGetOrAddSampler(scopeName));
+        }
+
+        // Creates a cached string named GPU + CPU scope into a command buffer
+        public static ProfilingScope GetGpuCpuScope(CommandBuffer cmd, string scopeName)
+        {
+            return new ProfilingScope(cmd, TryGetOrAddSampler(scopeName));
+        }
+
+        // Creates a cached "immediate" CPU scope using a reflected method
+        public static ProfilingScope GetCpuScope(System.Reflection.MethodBase method)
+        {
+            return new ProfilingScope(null, TryGetOrAddSampler(method));
+        }
+
+        // Creates a cached GPU + CPU scope into a command buffer using a reflected method
+        public static ProfilingScope GetGpuCpuScope(CommandBuffer cmd, System.Reflection.MethodBase method)
+        {
+            return new ProfilingScope(cmd, TryGetOrAddSampler(method));
+        }
+
+        // Creates a cached CPU "immediate" scope using a a generic hash key and two-part dynamic name of form "categoryName.detailName".
+        public static ProfilingScope GetCpuScope(int hashKey, string categoryName, string detailName)
+        {
+            return new ProfilingScope(null, TryGetOrAddSampler(hashKey, categoryName, detailName));
+        }
+
+        // Creates a cached GPU + CPU command-buffer scope using a generic hash key and two-part dynamic name of form "categoryName.detailName".
+        public static ProfilingScope GetGpuCpuScope(CommandBuffer cmd, int hashKey, string categoryName, string detailName)
+        {
+            return new ProfilingScope(cmd, TryGetOrAddSampler(hashKey, categoryName, detailName));
+        }
```

Comment:
I strongly disagree. The purpose is to make the API  clear and easy to use oneliners that are easy switch between the naming source. This is also an effect of language limits. All my attempts with lambdas failed as they allocate and allocation avoidance introduces a lot of boilerplate. However, ideally these would be part of the ProfilingScope itself, but that would need to be another PR as it touches other teams.

Reply:
Now that I think about this with current state of the code and knowledge that Reflection fails for IL2CPP. This could be just a simple additive change to add static factory functions to the ProfilingScope with no real impact on current code. Realistically that would be the string, enum and static sampler variants.
And the URP custom stuff would need to have a function here or just suffer the boilerplate way of building from multiple parts. Which might not be that many cases.

Reply:
Fixed.